corehq/apps/users/models.py

```diff
@@ -1334,6 +1334,13 @@ def set_location(self, location):
         Set the location, and all important user data, for
         the user.
         """
+        if not location:
+            # Remove associated info
+            self.user_data.pop('commtrack-supply-point', None)
+            self.user_data.pop('commcare_primary_case_sharing_id', None)
+            self.location_id = None
+            return
+
```

Comment:
It only does that if you do `user.set_location(None)`, in which case it basically undoes whatever happens when you set a true location.